### PR TITLE
[State Sync] Add simple state sync request rate limiting.

### PR DESCRIPTION
**Pull request by JoshLind**
## Description
This PR adds a configurable state sync request rate limiter to the storage service. Using this, we can rate limit the number of requests per second (per peer) to the storage service. This is useful for responding to malicious peers that may attempt to make many state sync requests to slow down the node.

Note: the rate limit currently only applies to peers on the `public` network, as these are untrusted. 

## Testing Plan
New and existing test infrastructure.

---

> [!NOTE]
> **Medium Risk**
> Introduces new request-throttling behavior and a new error path in the state sync storage service; misconfiguration could unintentionally reject legitimate public-network traffic, though the feature is disabled by default.
> 
> **Overview**
> Adds an optional **per-peer request rate limiter** for the state sync storage service, configurable via `StorageServiceConfig.max_requests_per_second_per_peer` (disabled by default and only enforced for `public` network peers).
> 
> The server now tracks per-peer token buckets in `RequestModerator`, returns a new `TooManyRequests` error mapped through to `StorageServiceError`, and exports a `STORAGE_REQUESTS_RATE_LIMITED` metric; rate-limit state is garbage-collected for disconnected peers and covered by new `rate_limiter` tests. Separately, `aptos-token-bucket` relaxes constructor assertions and adds handling/tests for zero refill-rate edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 87db276333dad133daed678f35697e50592ed460. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: state-sync/storage-service/server/src/moderator.rs [hunk 1]
```diff
@@ -138,6 +142,23 @@ impl RequestModerator {
...
+                    request.get_label(),
+                );
+
+                // Return an error
+                return Err(Error::TooManyRequests(format!(
+                    "Peer has exceeded the request rate limit ({:?} req/s). Request: {}",
+                    self.storage_service_config.max_requests_per_second_per_peer,
+                    request.get_label()
+                )));
+            }
```
[the remark this answers is not in the document]
Comment (JoshLind) [reply state-sync/storage-service/server/src/moderator.rs:159]: This is fine. We're rate limiting requests and storage reads per user. It makes sense to count additional storage lookups (e.g., for optimistic fetches) against the peer that requested it.

File: state-sync/storage-service/server/src/moderator.rs [hunk 2]
```diff
@@ -106,6 +108,7 @@ pub struct RequestModerator {
     aptos_data_client_config: AptosDataClientConfig,
     cached_storage_server_summary: Arc<ArcSwap<StorageServerSummary>>,
     peers_and_metadata: Arc<PeersAndMetadata>,
+    peer_rate_limit_states: Arc<DashMap<PeerNetworkId, Mutex<TokenBucket>>>,
```
Comment (zekun000) [inline state-sync/storage-service/server/src/moderator.rs]: I don't think you need a Mutex for values inside DashMap?

Comment (JoshLind) [reply state-sync/storage-service/server/src/moderator.rs]: Aah, thanks! (AI both made that mistake and fixed it) 😆

**Comment by zekun000** [comment]
some test failure?

**Comment by JoshLind** [comment]
> some test failure?

@zekun000, looks like it was a flake (unrelated to this PR 😄). Should be good to go!
